src/web/app/pages-instructor/instructor-comments.component.ts

```diff
@@ -0,0 +1,132 @@
+import { FeedbackResponseCommentService } from '../../services/feedback-response-comment.service';
+import { StatusMessageService } from '../../services/status-message.service';
+import {
+    FeedbackResponseComment,
+    FeedbackSession, FeedbackSessionPublishStatus, FeedbackSessionSubmissionStatus,
+    ResponseVisibleSetting,
+    SessionVisibleSetting,
+} from '../../types/api-output';
+import { Intent } from '../../types/api-request';
+import { CommentRowModel } from '../components/comment-box/comment-row/comment-row.component';
+import { CommentTableModel } from '../components/comment-box/comment-table/comment-table.component';
+import { CommentToCommentRowModelPipe } from '../components/comment-box/comment-to-comment-row-model.pipe';
+import { ErrorMessageOutput } from '../error-message-output';
+
+/**
+ * Base class for instructor comment CRUD operations.
+ */
+export abstract class InstructorCommentsComponent {
+
+  session: FeedbackSession = {
+    courseId: '',
+    timeZone: '',
+    feedbackSessionName: '',
+    instructions: '',
+    submissionStartTimestamp: 0,
+    submissionEndTimestamp: 0,
+    gracePeriod: 0,
+    sessionVisibleSetting: SessionVisibleSetting.AT_OPEN,
+    responseVisibleSetting: ResponseVisibleSetting.AT_VISIBLE,
+    submissionStatus: FeedbackSessionSubmissionStatus.OPEN,
+    publishStatus: FeedbackSessionPublishStatus.NOT_PUBLISHED,
+    isClosingEmailEnabled: true,
+    isPublishedEmailEnabled: true,
+    createdAtTimestamp: 0,
+  };
+  currInstructorName?: string;
+
+  // this is a separate model for instructor comments
+  // from responseID to comment table model
+  instructorCommentTableModel: Record<string, CommentTableModel> = {};
+
+  protected constructor(
```

Comment:
Since the class is `abstract`, the keyword `protected` for constructor is optional.

Reply:
I'm gonna leave it there then if that's alright, cause otherwise my IDE shows a red squiggly line :P